src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/NonResolvableKeyDirectiveTests.cs

```diff
@@ -0,0 +1,325 @@
+                .Name("Query")
+                .Field("someField")
+                .Argument("a", a => a.Type<IntType>())
+                .Type("ITestType")
+                .Resolve(_ => new { Id = 1, Name = "bar" })
+            )
+            .AddInterfaceType(
+                o =>
+                {
+                    o.Name("ITestType")
+                        .Key("id", resolvable: false);
+                    o.Field("id")
+                        .Type<IntType>();
+                    o.Field("name")
+                        .Type<StringType>();
+                })
+            .AddObjectType(
+                o =>
+                {
+                    o.Name("TestTypeResolvableKey")
+                        .Implements("ITestType")
+                        .Key("id");
+                    o.Field("id")
+                        .Type<IntType>()
+                        .Resolve(_ => 1);
+                    o.Field("name")
+                        .Type<StringType>()
+                        .Resolve(_ => "bar");
+                })
+            .AddObjectType(
+                o =>
+                {
+                    o.Name("TestType")
+                        .Implements("ITestType")
+                        .Key("id", resolvable: false);
+                    o.Field("id")
+                        .Type<IntType>()
+                        .Resolve(_ => 1);
+                    o.Field("name")
+                        .Type<StringType>()
+                        .Resolve(_ => "bar");
+                })
+            .BuildSchemaAsync();
+
+        // act
+        var testType = schema.GetType<InterfaceType>("ITestType");
+
+        // assert
+        Assert.Collection(
+            testType.Directives,
+            item =>
+            {
+                Assert.Equal(FederationTypeNames.KeyDirective_Name, item.Type.Name);
+                var syntaxNode = item.AsSyntaxNode();
+                Assert.Equal("fields", syntaxNode.Arguments[0].Name.ToString());
+                Assert.Equal("\"id\"", syntaxNode.Arguments[0].Value.ToString());
+                Assert.Equal("resolvable", syntaxNode.Arguments[1].Name.ToString());
+                Assert.Equal("false", syntaxNode.Arguments[1].Value.ToString());
+            });
+
+        schema.ToString().MatchSnapshot();
+    }
+
+    public class Query<T>
+    {
+        // ReSharper disable once InconsistentNaming
+        public T someField(int id) => default!;
+    }
+
+    [Key("id", resolvable: false)]
+    public class TestTypeClassDirective : ITestTypeInterfaceDirective
+    {
+        public int Id { get; set; }
+    }
+
+    [Key("id", resolvable: false)]
+    public interface ITestTypeInterfaceDirective
+    {
+        int Id { get; }
+    }
+
+    public class TestTypePropertyDirective
+    {
+        [Key(null!, resolvable: false)]
+        public int Id { get; set; }
+    }
+
+    public class TestTypePropertyDirectives
+    {
+        [Key(null!, resolvable: false)]
```

Comment:
Is this a desirable syntax? Should resolvable be unsupported when specified on a field set?

A question for maintainers/Apollo

Reply:
Hi @michaelstaib, if you could have a small think about this comment, it would be appreciated. Once decision has been made/changes implemented, the rest can be reviewed a bit later. Thanks!